src/components/cylc/cylcObject/Menu.vue

```diff
@@ -151,14 +154,22 @@ export default {
       }
       return !!this.$workflowService.primaryMutations[this.type]
     },
-
+    ...mapState('user', ['user']),
     displayMutations () {
-      if (!this.mutations) {
+      if (!this.mutations || this.user.permissions.length < 2) {
         return []
       }
       const shortList = this.$workflowService.primaryMutations[this.type]
+
+      const mutsWithAuth = this.mutations.map(
+        (x) => {
+          if (this.user.permissions.map((str) => str.toLowerCase()).includes(x[0].name.toLowerCase())) {
```

Comment:
See comments above; I think here every time `this.mutations` or `this.user` changes, this `displayMutations` computed property will be re-evaluated. In this part, the `this.user.permissions.map` will be executed for each mutation. Using other computer property instead, I think it could be simpler to read the relationship between reactive properties in Vue, and would avoid re-calculating the `this.user.permissions.map((str) => str.toLowerCase())`.

But I don't trust my code after 4-5 PM and without :coffee: , and the PR is still in draft... but just in case you'd like to have fun experimenting more with the Vue reactivity system @datamel :tada: